scheduler/scheduler-smartproxy-common/src/main/java/org/ow2/proactive/scheduler/smartproxy/common/AbstractSmartProxy.java

```diff
@@ -611,31 +598,27 @@ protected String prepareJobOutput(TaskFlowJob job, String localOutputFolder, Str
      * @param job
      * @param localInputFolder path to the input folder on local machine if null, no actions
      *                         will be performed concerning the input data for this job
-     * @param pushURL          the url where input data is to be pushed before the job
+     * @param pushURLs         the urls where input data is to be pushed before the job
      *                         submission
      * @param newFolderName    name of the new folder to be created
      * @return String representing the updated value of the push_url
      */
-    protected String prepareJobInput(Job job, String localInputFolder, String pushURL, String newFolderName)
+    protected String prepareJobInput(Job job, String localInputFolder, List<String> pushURLs, String newFolderName)
             throws NotConnectedException, PermissionException {
-        // if the job defines an input space
-        // and the localInputFolder is not null
         // create a remote folder for the input data
         // and update the InputSpace property of the job to reference that
         // folder
-
-        String inputSpace_url = job.getInputSpace();
         String push_url_updated = "";
 
         // the input folder, on the remote input space, relative to the root url
         String inputFolder = "";
-        if ((localInputFolder != null) && (inputSpace_url != null) && (!inputSpace_url.equals("")) &&
-            (pushURL != null)) {
+        if ((localInputFolder != null) && (pushURLs != null) && (pushURLs.size() > 0)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)